### PR TITLE
feat: add metadata from first paragraph and heading

### LINKED ISSUE #1903: Use the first paragraph of markdown file as meta description
Evidence currently allows you to specify the [meta description](https://moz.com/learn/seo/meta-description) for the page using [frontmatter](https://docs.evidence.dev/reference/markdown/#frontmatter).

If the user does not specify this explicitly, we should automatically use the first paragraph to generate this content.

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
### Description

close #1903 

Changes:
- Added pre-processor to extract first h1-heading and paragraph from svelte markup to be used as page meta title and description, if frontmatter title / description is missing.

### Checklist

- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] I have added to the docs where applicable
- [ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable

### CODE REVIEW

File: .changeset/selfish-ducks-sell.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+'@evidence-dev/db-orchestrator': minor
+'@evidence-dev/plugin-connector': minor
+'@evidence-dev/preprocess': minor
+'@evidence-dev/query-store': minor
+'@evidence-dev/sdk': minor
+'@evidence-dev/universal-sql': minor
+'@evidence-dev/core-components': minor
+'my-evidence-project': minor
+'@evidence-dev/components': minor
+'evidence-test-environment': minor
```
Comment (ItsMeBrianD) [inline .changeset/selfish-ducks-sell.md:16]: This should only be a `patch` change for the impacted package (in this case `preprocess`)

File: packages/lib/preprocess/index.cjs [hunk 2]
```diff
@@ -32,7 +33,7 @@ module.exports = function evidencePreprocess(componentDevelopmentMode = false) {
 						'*': 'markdown'
 					}
 				],
-				[rehypeSlug],
+				rehypeSlug,
```
Comment (ItsMeBrianD) [inline packages/lib/preprocess/index.cjs:36]: Is this change related?

**Review by ItsMeBrianD — CHANGES_REQUESTED** [review]

File: packages/lib/preprocess/index.cjs (hunk 2 above)
Comment (AyushAgrawal-A2) [reply packages/lib/preprocess/index.cjs:36]: @ItsMeBrianD 
square brackets are required to supply options at index 1, which we are not supplying here
ref: https://mdsvex.pngwn.io/docs#remarkplugins--rehypeplugins

File: .changeset/selfish-ducks-sell.md (hunk 1 above)
Comment (AyushAgrawal-A2) [reply .changeset/selfish-ducks-sell.md:16]: @ItsMeBrianD 
sure will change it, shall I also revert the pnpm.lock file commit.
Without commiting pnpm.lock file, previews were not getting deployed.

Comment (ItsMeBrianD) [reply .changeset/selfish-ducks-sell.md:16]: No the lock file is okay to include!

Comment (AyushAgrawal-A2) [reply .changeset/selfish-ducks-sell.md]: Done

**Comment by ItsMeBrianD** [comment]
With an upcoming change to how we are going to do preprocessing; I'm going to close this, but have noted it in #2113
